lib/cylc/task_events_mgr.py

```diff
@@ -400,11 +401,17 @@ def process_message(
             self.pflag = True
             itask.state.reset_state(TASK_STATUS_SUBMITTED)
             self._reset_job_timers(itask)
-        elif an_output_was_satisfied:
+            # We should really have a special 'vacated' handler, but given that
+            # this feature can only be used on the deprecated loadleveler
+            # system, we should probably aim to remove support for job vacation
+            # instead. Otherwise, we should have:
+            # self.setup_event_handlers(itask, 'vacated', message)
+        elif completed_trigger:
             # Message of an as-yet unreported custom task output.
             # No state change.
             self.pflag = True
             self.suite_db_mgr.put_update_task_outputs(itask)
+            self.setup_event_handlers(itask, completed_trigger, message)
```

Comment:
This allows known outputs to be used as events (for generic handler list only).

https://github.com/cylc/cylc/issues/2806#issuecomment-430589075

If desirable, I'll document the new capability.
If not desirable, I'll revert.

Reply:
I like it. :+1:

Reply:
In which case, I'll add some basic docs for this. Note this is likely to make #2756 a bit more difficult!